static/src/javascripts/projects/common/modules/commercial/creatives/hosted-thrasher.js

```diff
@@ -0,0 +1,41 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/template',
+    'common/modules/commercial/hosted-video',
+    'common/modules/commercial/creatives/add-tracking-pixel',
+    'text!common/views/commercial/creatives/hosted-thrasher.html'
+], function (
+    fastdom,
+    template,
+    hostedVideo,
+    addTrackingPixel,
+    hostedVideoStr
+) {
+    var hostedVideoTpl;
+
+    var HostedThrasher = function ($adSlot, params) {
```

Comment:
@jamesgorrie You can extend the prototype of any function - all JavaScript functions are objects:

```javascript
function Foo() {
    this.x = 1;
}

Foo.prototype.y = 2;
```